src/components/subtitlesync/subtitlesync.js

```diff
@@ -1,125 +1,138 @@
-        setTimeout(function () {
-            subtitleSyncSlider.enableKeyboardDragging();
-        }, 0);
-    }
-
-    subtitleSyncContainer.classList.add('hide');
-
-    subtitleSyncTextField.updateOffset = function (offset) {
-        this.textContent = offset + 's';
-    };
-
-    subtitleSyncTextField.addEventListener('click', function () {
-        // keep focus to prevent fade with osd
-        this.hasFocus = true;
-    });
-
-    subtitleSyncTextField.addEventListener('keydown', function (event) {
-        if (event.key === 'Enter') {
-            // if input key is enter search for float pattern
-            let inputOffset = /[-+]?\d+\.?\d*/g.exec(this.textContent);
-            if (inputOffset) {
-                inputOffset = inputOffset[0];
-                inputOffset = parseFloat(inputOffset);
-
-                subtitleSyncSlider.updateOffset(inputOffset);
-            } else {
-                this.textContent = (playbackManager.getPlayerSubtitleOffset(player) || 0) + 's';
-            }
-            this.hasFocus = false;
-            event.preventDefault();
-        } else {
-            // keep focus to prevent fade with osd
-            this.hasFocus = true;
-            if (event.key.match(/[+-\d.s]/) === null) {
-                event.preventDefault();
-            }
-        }
+// Constants
+const DEFAULT_OFFSET = 0;
+
+function createSliderBubbleHtml(_, value) {
+    return '<h1 class="sliderBubbleText">'
+        + (value > 0 ? '+' : '') + parseFloat(value) + 's'
+        + '</h1>';
+}
 
-        // FIXME: TV layout will require special handling for navigation keys. But now field is not focusable
-        event.stopPropagation();
-    });
+class OffsetController {
+    constructor(player, slider, textField) {
+        this.player = player;
+        this.slider = slider;
+        this.textField = textField;
 
-    subtitleSyncTextField.blur = function () {
-        // prevent textfield to blur while element has focus
-        if (!this.hasFocus && this.prototype) {
-            this.prototype.blur();
-        }
-    };
+        this._initSlider();
+        this._initTextField();
 
-    function updateSubtitleOffset() {
-        const value = parseFloat(subtitleSyncSlider.value);
-        // set new offset
-        playbackManager.setSubtitleOffset(value, player);
-        // synchronize with textField value
-        subtitleSyncTextField.updateOffset(value);
+        // Set initial offset
+        this.reset();
     }
 
-    subtitleSyncSlider.updateOffset = function (sliderValue) {
-        // default value is 0s = 0ms
-        this.value = sliderValue === undefined ? 0 : sliderValue;
+    get currentOffset() {
+        return parseFloat(this.slider.value);
+    }
 
-        updateSubtitleOffset();
-    };
+    set currentOffset(value) {
+        this.slider.value = value.toString();
+        playbackManager.setSubtitleOffset(value, this.player);
```

Comment:
```suggestion
        this.slider.value = value.toString();

        // rely on slider value trimming
        value = this.currentOffset;

        playbackManager.setSubtitleOffset(value, this.player);
```
Re-assign value to fix floating point rounding errors.
![float-1](https://github.com/user-attachments/assets/048ce16b-7366-4789-9fee-760582bb70cd)
Previously, it was `const value = parseFloat(subtitleSyncSlider.value);`.
You may want to rephrase the comment.